pkg/cache/clusterqueue.go

```diff
@@ -493,3 +516,50 @@ func resetUsage(lqUsage FlavorResourceQuantities, cqUsage FlavorResourceQuantiti
 func workloadBelongsToLocalQueue(wl *kueue.Workload, q *kueue.LocalQueue) bool {
 	return wl.Namespace == q.Namespace && wl.Spec.QueueName == q.Name
 }
+
+// RequestableCohortQuota recalculates cohort.RequestableResources when LendingLimit enabled,
+// because when borrowing from the cohort, need to count cq's own nominal quota in.
+func (c *ClusterQueue) RequestableCohortQuota(fName kueue.ResourceFlavorReference, rName corev1.ResourceName) (val int64) {
+	if c.Cohort.RequestableResources == nil {
+		return 0
+	}
+	if c.Cohort.RequestableResources[fName] == nil {
+		return 0
+	}
+	requestableCohortQuota := c.Cohort.RequestableResources[fName][rName]
+
+	// When feature LendingLimit enabled, RequestableResources only contains the sum of
+	// all CQs' LendingLimit, so we need to add the guaranteed resources.
```

Comment:
```suggestion
	// When feature LendingLimit enabled, cohort.requestableResource accumulated the lendingLimit if not null
	// rather than the flavor's quota, then the total available quota should include its own guaranteed resources.
```